web/src/features/comments/CommentList.tsx

```diff
@@ -0,0 +1,262 @@
+import {
+  Avatar,
+  AvatarFallback,
+  AvatarImage,
+} from "@/src/components/ui/avatar";
+import { Button } from "@/src/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+} from "@/src/components/ui/form";
+import { Textarea } from "@/src/components/ui/textarea";
+import { useHasProjectAccess } from "@/src/features/rbac/utils/checkProjectAccess";
+import { api } from "@/src/utils/api";
+import { cn } from "@/src/utils/tailwind";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { type CommentObjectType, CreateCommentData } from "@langfuse/shared";
+import { ArrowUpToLine, LoaderCircle, Trash } from "lucide-react";
+import { useSession } from "next-auth/react";
+import React, { useEffect, useMemo } from "react";
+import { useForm } from "react-hook-form";
+import { type z } from "zod";
+
+const formatCommentTimestamp = (timestamp: Date): string => {
+  const diffInMs = new Date().getTime() - timestamp.getTime();
+  const diffInMinutes = diffInMs / (1000 * 60);
+  const diffInHours = diffInMinutes / 60;
+  const diffInDays = diffInHours / 24;
+
+  if (diffInHours < 1) {
+    return `${Math.floor(diffInMinutes)} minutes ago`;
+  } else if (diffInHours < 24) {
+    return `${Math.floor(diffInHours)} hours ago`;
+  } else if (diffInDays < 7) {
+    return `${Math.floor(diffInDays)} days ago`;
+  } else {
+    return timestamp.toLocaleDateString("en-US", {
+      year: "2-digit",
+      month: "numeric",
+      day: "numeric",
+    });
+  }
+};
+
+export function CommentList({
+  projectId,
+  objectId,
+  objectType,
+  isVisible,
```

Comment:
Is `isVisible` necessary? Just tried removing the prop, the tRPC route was still only executed when the drawer was open. I believe it can be removed.